xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1056 @@
+        super().__init__(
+            operands=[tile],
+            attributes={"sym_name": sym_name},
+            result_types=[buffer_type],
+        )
+
+
+@irdl_op_definition
+class TileOp(IRDLOperation):
+    name = "AIE.tile"
+    col: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    row: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    result: OpResult = result_def(IndexType())
+
+    def __init__(self, col: IntegerAttr[i32], row: IntegerAttr[i32]):
+        super().__init__(
+            attributes={"col": col, "row": row}, result_types=[IndexType()]
+        )
+
+
+@irdl_op_definition
+class ConnectOp(IRDLOperation):
+    name = "AIE.connect"
+    sourceBundle: WireBundleAttr = attr_def(WireBundleAttr)
+    sourceChannel: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    destBundle: WireBundleAttr = attr_def(WireBundleAttr)
+    destChannel: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+
+    def __init__(
+        self,
+        sourceBundle: WireBundleAttr,
+        sourceChannel: IntegerAttr[i32],
+        destBundle: WireBundleAttr,
+        destChannel: IntegerAttr[i32],
+    ):
+        super().__init__(
+            attributes={
+                "sourceBundle": sourceBundle,
+                "sourceChannel": sourceChannel,
+                "destBundle": destBundle,
+                "destChannel": destChannel,
+            }
+        )
+
+    def print(self, printer: Printer):
+        printer.print(
+            "<",
+            self.sourceBundle.data,
+            ": ",
+            self.sourceChannel.value.data,
+            ", ",
+            self.destBundle.data,
+            ": ",
+            self.destChannel.value.data,
+            ">",
+        )
+
+
+@irdl_op_definition
+class CoreOp(IRDLOperation):
+    name = "AIE.core"
+    stackSize: IntegerAttr[i32] = attr_def(IntegerAttr)
+    tile: Operand = operand_def(IndexType())
+    region: Region = region_def()
+    link_with: StringAttr = opt_attr_def(StringAttr)
+    result: OpResult = result_def(IndexType())
+
+    def __init__(
+        self,
+        stackSize: IntegerAttr[i32],
+        tile: IndexType(),
+        region: Region,
+        link_with: StringAttr | None = None,
+    ):
+        super().__init__(
+            attributes={"stackSize": stackSize, "link_with": link_with},
+            operands=[tile],
+            regions=[region],
+            result_types=[IndexType()],
+        )
+
+    def print(self, printer: Printer):
+        printer.print(" (")
+        printer.print_operand(self.tile)
+        printer.print(") ")
+        printer.print_region(self.region)
+        if self.link_with is not None:
+            printer.print(' { link_with="', self.link_with, '" }')
+
+
+@irdl_op_definition
+class DMABDOp(IRDLOperation):
+    name = "AIE.dmaBd"
+    offset: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    length: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    AB: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
```

Comment:
nit, we've been trying to stick with lowercase names for our python classes, even if that disagrees with the way that's printed in the IR:
```suggestion
    ab: IntegerAttr[i32] = attr_def(IntegerAttr[i32], attr_name="AB")
```

Reply:
Also, the type annotation isn't actually necessary, I'm guilty of having started this before realising that only the thing after the `=` is actually needed
```suggestion
    ab = attr_def(IntegerAttr[i32], attr_name="AB")
```